data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/CsaProfileRemedialActionsCreator.java

```diff
@@ -0,0 +1,222 @@
+                        checkContingencyAndFillImportedCoIds(faraoContingenciesIds, contingencyWithRemedialActionPropertyBag, parentRemedialActionPropertyBag, remedialActionId, randomCombinationConstraintKind);
+                    }
+
+                    addOnContingencyStateUsageRules(networkActionAdder, faraoContingenciesIds, randomCombinationConstraintKind);
+                } else { // no contingency linked to RA --> on instant case
+                    String kind = parentRemedialActionPropertyBag.get(CsaProfileConstants.RA_KIND);
+                    if (kind.equals(CsaProfileConstants.RemedialActionKind.PREVENTIVE.toString())) {
+                        networkActionAdder.newOnInstantUsageRule().withUsageMethod(UsageMethod.AVAILABLE).withInstant(Instant.PREVENTIVE).add();
+                    } else {
+                        networkActionAdder.newOnInstantUsageRule().withUsageMethod(UsageMethod.AVAILABLE).withInstant(Instant.CURATIVE).add();
+                    }
+                }
+
+                networkActionAdder.add();
+                csaProfileRemedialActionCreationContexts.add(CsaProfileRemedialActionCreationContext.imported(remedialActionId, remedialActionId, targetRemedialActionNameOpt.orElse(remedialActionId), "", false));
+
+            } catch (FaraoImportException e) {
+                csaProfileRemedialActionCreationContexts.add(CsaProfileRemedialActionCreationContext.notImported(remedialActionId, e.getImportStatus(), e.getMessage()));
+            }
+        }
+        this.cracCreationContext.setRemedialActionCreationContext(csaProfileRemedialActionCreationContexts);
+    }
+
+    private void addOnContingencyStateUsageRules(NetworkActionAdder networkActionAdder, List<String> faraoContingenciesIds, String randomCombinationConstraintKind) {
+        faraoContingenciesIds.forEach(faraoContingenciesId -> {
+            OnContingencyStateAdder<NetworkActionAdder> onContingencyStateAdder = networkActionAdder.newOnContingencyStateUsageRule().withInstant(Instant.CURATIVE).withContingency(faraoContingenciesId);
+
+            if (randomCombinationConstraintKind.equals(CsaProfileConstants.ElementCombinationConstraintKind.INCLUDED.toString())) {
+                onContingencyStateAdder.withUsageMethod(UsageMethod.FORCED).add();
+
+            }
+            if (randomCombinationConstraintKind.equals(CsaProfileConstants.ElementCombinationConstraintKind.CONSIDERED.toString())) {
+                onContingencyStateAdder.withUsageMethod(UsageMethod.AVAILABLE).add();
+            }
+            if (randomCombinationConstraintKind.equals(CsaProfileConstants.ElementCombinationConstraintKind.EXCLUDED.toString())) {
+                onContingencyStateAdder.withUsageMethod(UsageMethod.UNAVAILABLE).add();
+                networkActionAdder.newOnInstantUsageRule().withUsageMethod(UsageMethod.AVAILABLE).withInstant(Instant.CURATIVE).add();
```

Comment:
this may happen multiple times. maybe add at least a TODO to make it more optimal